src/components/Resources/ResourceList.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import ResourceCard from './ResourceCard';
+import styles from './Resource.module.css';
+
+function ResourceList() {
+  const [resources, setResources] = useState([]);
+  const [tagMap, setTagMap] = useState(null); // null until loaded
+  const [status, setStatus] = useState('loading'); //loading, failed, succeeded
+  const [visibleArticles, setVisibleArticles] = useState(6);
+
+  useEffect(() => {
+    async function fetchData() {
+      setStatus('loading')
+      try {
+        const [resourcesRes, tagsRes] = await Promise.all([
+          fetch('https://seshatbe.up.railway.app/resources'),
+          fetch('https://seshatbe.up.railway.app/tags'),
+        ]);
+
+        if (!resourcesRes.ok || !tagsRes.ok) {
+        throw new Error('Fetch failed');
+      }
+
+        const resourcesData = await resourcesRes.json();
+        const tagsData = await tagsRes.json();
+
+        // Convert tags into a map using string keys
+        const tagMapObj = {};
+        tagsData.forEach(tag => {
+          tagMapObj[String(tag.id)] = tag.tag;
+        });
+
+        setResources(resourcesData);
+        setTagMap(tagMapObj); // set AFTER map is ready
+        setStatus('succeeded');
+      }
+
+      catch (error) {
+        console.error('Failed to fetch resources or tags:', error);
+        setStatus('failed');
+      }
+    }
+
+    fetchData();
+  }, []);
+
+  if (status === 'failed') {
+    return (
+      <div className={styles.retry}>
+        <h2>Failed to load resources.</h2>
+        <p>Please try again later.</p>
+        <button onClick={() => window.location.reload()}>Retry</button>
```

Comment:
It would be better to use `fetchData()` here rather than `window.location.reload()`. Reloading the page is slower and will clear all page state.

```suggestion
        <button onClick={() => fetchData()}>Retry</button>
```

For this to work you would have move the declaration of `fetchData()` out of `useEffect`.